src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/MemoryCache.cs

```diff
@@ -62,11 +62,11 @@ public void Set(TKey key, TValue value)
 
         protected virtual void Compact()
         {
-            var kvps = _dict.OrderBy(x => x.Value.LastAccess);
+            var kvps = _dict.OrderBy(x => x.Value.LastAccess).ToArray();
 
             for (var i = 0; i < _sizeLimit / 2; i++)
             {
-                _dict.Remove(kvps.ElementAt(i).Key);
```

Comment:
>ElementAt [](start = 34, length = 9)

I see a couple other ElementAt calls throughout the razor codebase that should probably be evaluated.

Reply:
Ya i'll do a run through and do a follow up if needed